### ISSUE TITLE
Explore/Elastic: Query results are processed as logs when user switches mode from Logs to Metrics

### ISSUE THREAD

**Issue by ivanahuckova** [grafana/grafana#24077]
**What happened**:
In play, with Elastic datasource, when we run logs query first and then switch to metrics, the query results are still processed as logs results (even though it should be processed as time series results)

**What you expected to happen**:
See results of metrics query.

**How to reproduce it (as minimally and precisely as possible)**:
1. Go to play with [Elastic Metrics as a data source and run logs query]( https://play.grafana.org/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Elastic%20Metrics%22,%7B%22datasource%22:%22Elastic%20Metrics%22,%22query%22:%22%22,%22isLogsQuery%22:true%7D,%7B%22mode%22:%22Logs%22%7D,%7B%22ui%22:%5Btrue,true,true,%22none%22%5D%7D%5D)
2. Click on Metrics button, see in the url `isLogsQuery: true` (we also see error due to https://github.com/grafana/grafana/pull/24294 but it is not relevant to this PR). 

**Environment**:
- Grafana version: master (9939d3b)

**Comment by aocenas** [comment]
I think elastic metrics data source on play site is not setup for logs and we have separate elastic logs data source. Not great, at least we should show some better error message or something.

**Comment by ivanahuckova** [comment]
I have looked into the issue and found out following:
For elastic data source, we have 2 types of query editor components. Logs editor is a [react component](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/elasticsearch/components/ElasticsearchQueryField.tsx) an metrics editor is a [react component that renders angular component](https://github.com/grafana/grafana/blob/master/public/app/features/explore/QueryEditor.tsx). When we switch the mode, we switch the editors. In logs editor, we are setting `isLogsQuery: true` [to each query](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/elasticsearch/components/ElasticsearchQueryField.tsx#L52). And when we switch mode, `isLogsQuery: true` stays. So we need to change `isLogsQuery: false` when we switch Explore mode to metrics. This way, results are going to be processed as a metrics query and not as a logs query. 

I have unsuccessfully tried to change `isLogsQuery: false` in componentWillUnmount in logs editor. It didn't work as metrics editor received query props before the `isLogsQuery: false` changed. So another solution would be to update `isLogsQuery: false` when we are mounting metrics editor. So the solution (which I am not sure is ideal, but after the discussion, we couldn't think of anything else), would be to pass information about explore mode to that elastic editor, and change `isLogsMode = false` if:
- we are in Explore
- ExploreMode is Metrics
- isLogsQuery === true

I have tried this, but as I don't have much angular experience and this is currently not a priority, I have decided to prioritise other issues.
